ToolBox/CoinApi/DefaultConnectionHandler.cs

```diff
@@ -0,0 +1,196 @@
+                {
+                    _lastDataReceivedTime = DateTime.UtcNow;
+                }
+
+                try
+                {
+                    while (!_cancellationTokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(1000);
+
+                        if (!_isEnabled) continue;
+
+                        try
+                        {
+                            TimeSpan elapsed;
+                            lock (_lockerConnectionMonitor)
+                            {
+                                elapsed = DateTime.UtcNow - _lastDataReceivedTime;
+                            }
+
+                            if (!_connectionLost && elapsed > _maximumIdleTimeSpan)
+                            {
+                                _connectionLost = true;
+                                nextReconnectionAttemptUtcTime = DateTime.UtcNow.AddSeconds(nextReconnectionAttemptSeconds);
+
+                                OnConnectionLost();
+                            }
+                            else if (_connectionLost)
+                            {
+                                if (elapsed <= _maximumIdleTimeSpan)
+                                {
+                                    _connectionLost = false;
+                                    nextReconnectionAttemptSeconds = _minimumSecondsForNextReconnectionAttempt;
+
+                                    OnConnectionRestored();
+                                }
+                                else
+                                {
+                                    if (DateTime.UtcNow > nextReconnectionAttemptUtcTime)
+                                    {
+                                        // double the interval between attempts (capped to 1 minute)
+                                        nextReconnectionAttemptSeconds = Math.Min(nextReconnectionAttemptSeconds * 2, _maximumSecondsForNextReconnectionAttempt);
+                                        nextReconnectionAttemptUtcTime = DateTime.UtcNow.AddSeconds(nextReconnectionAttemptSeconds);
+
+                                        OnReconnectRequested();
+                                    }
+                                }
+                            }
+                        }
+                        catch (Exception exception)
+                        {
+                            Log.Error($"Error in DefaultConnectionHandler: {exception}");
+                        }
+                    }
+                }
+                catch (Exception exception)
+                {
+                    Log.Error(exception);
+                }
+            }) { IsBackground = true };
+
+            _connectionMonitorThread.Start();
+
+            waitHandle.WaitOne();
+        }
+
+        /// <summary>
+        /// Enables/disables monitoring of the connection
+        /// </summary>
+        public void EnableMonitoring(bool isEnabled)
```

Comment:
Function is called `EnableMonitoring` but it looks like it's intended to enable _or_ disable monitoring, however, if monitoring is already enabled then we won't disable monitor (`_isEnabled == true && isEnabled == false` seems like a broken case)?